judge/views/problem.py

```diff
@@ -213,8 +226,117 @@ def get_context_data(self, **kwargs):
+        if context['can_vote']:
+            vote = ProblemPointsVote.objects.filter(voter=user.profile, problem=self.object)
+        # whether or not they've already voted
+        context['has_voted'] = context['can_vote'] and vote.exists()
+        if context['has_voted']:
+            context['voted_points'] = vote.first().points  # the previous vote's points
+            context['voted_note'] = vote.first().note
+
+        if 'problem_points_vote_form' not in context:
+            context['problem_points_vote_form'] = ProblemPointsVoteForm({})
+
+        if 'has_errors' not in context:
+            context['has_errors'] = False
+
+        if 'points_placeholder' not in context:  # placeholder for the points
+            if context['has_voted']:  # if voted, the vote
+                context['points_placeholder'] = context['voted_points']
+            else:  # otherwise a *nice* default
+                context['points_placeholder'] = 69.42069
+
+        if 'note_placeholder' not in context:
+            if context['has_voted']:
+                context['note_placeholder'] = context['voted_note']
+            else:
+                context['note_placeholder'] = self.default_note()
+
+        all_votes = sorted([v.points for v in ProblemPointsVote.objects.filter(problem=self.object)])
+        context['has_votes'] = len(all_votes) > 0
+        if context['has_votes']:
+            context['mean_vote'] = sum(all_votes) / len(all_votes)
+            context['num_votes'] = len(all_votes)
+            context['min_vote'] = all_votes[0]
+            context['max_vote'] = all_votes[-1]
+
+            # provides index and value of the median of some range of the data
+            def median(left_index, right_index, data):
+                size = right_index - left_index + 1
+                index = left_index + (size - 1) / 2
+                value = None
+                if size % 2 == 1:
+                    value = data[int(index)]
+                else:
+                    value = (data[math.floor(index)] + data[math.ceil(index)]) / 2
+                return index, value
+
+            median_data = median(0, len(all_votes) - 1, all_votes)
+            context['median_vote'] = median_data[1]
+            median_index = median_data[0]
+
+            context['enough_data_for_plot'] = len(all_votes) > 2
+            if context['enough_data_for_plot']:
+                # box and whisker plot data
+                q1 = median(0, math.ceil(median_index - 1), all_votes)  # first quartile
+                q3 = median(math.floor(median_index + 1), len(all_votes) - 1, all_votes)  # second quartile
+                context['first_quartile'] = q1[1]
+                context['third_quartile'] = q3[1]
+
+            context['in_contest'] = contest_problem is not None
+
+            if context['can_vote']:
+                context['all_votes'] = all_votes
+
+            context['max_possible_vote'] = settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE
```

Comment:
Also check this against `settings.DMOJ_PROBLEM_MIN_PROBLEM_POINTS`

Reply:
Could you elaborate? I do not understand the issue.

Reply:
should be tagged to the line below (I think a force push somewhere messed this up)

Reply:
oh should I take the minimum or maximum of the two?

Reply:
Maximum. You shouldn't be able to vote for a lower point that can normally be set.